lib/Feature/FeatureModelWriter.cpp

```diff
@@ -0,0 +1,314 @@
+        BAD_CAST Feature1.getParent()->getName().data());
+    CHECK_RC
+  }
+
+  // children
+  RC = xmlTextWriterStartElement(Writer, XmlConstants::CHILDREN);
+  CHECK_RC
+  OrderedFeatureVector Children;
+  Children.insert(Feature1.children());
+  for (Feature *F : Children) {
+    RC = xmlTextWriterWriteElement(Writer, XmlConstants::OPTIONS,
+                                   BAD_CAST F->getName().data());
+    CHECK_RC
+  }
+  RC = xmlTextWriterEndElement(Writer); // CHILDREN
+  CHECK_RC
+
+  // exclude
+  RC = xmlTextWriterStartElement(Writer, XmlConstants::EXCLUDEDOPTIONS);
+  CHECK_RC
+  OrderedFeatureVector Exludes;
+  Exludes.insert(Feature1.excludes());
+  for (Feature *F : Exludes) {
+    RC = xmlTextWriterWriteElement(Writer, XmlConstants::OPTIONS,
+                                   BAD_CAST F->getName().data());
+    CHECK_RC
+  }
+  RC = xmlTextWriterEndElement(Writer); // EXCLUDEDOPTIONS
+  CHECK_RC
+
+  // implications
+  RC = xmlTextWriterStartElement(Writer, XmlConstants::IMPLIEDOPTIONS);
+  CHECK_RC
+  OrderedFeatureVector Implies;
+  Implies.insert(Feature1.implications());
+  for (Feature *F : Implies) {
+    RC = xmlTextWriterWriteElement(Writer, XmlConstants::OPTIONS,
+                                   BAD_CAST F->getName().data());
+    CHECK_RC
+  }
+  RC = xmlTextWriterEndElement(Writer); // IMPLIEDOPTIONS
+  CHECK_RC
+
+  // if numeric
+  if (NumericFeature::classof(&Feature1)) {
+    auto &Nf = dynamic_cast<NumericFeature &>(Feature1);
+    auto ValueVariant = Nf.getValues();
+    if (std::holds_alternative<std::pair<int, int>>(ValueVariant)) {
+      auto [Min, Max] = std::get<std::pair<int, int>>(ValueVariant);
+      RC = xmlTextWriterWriteElement(Writer, XmlConstants::MINVALUE,
+                                     BAD_CAST std::to_string(Min).data());
+      CHECK_RC
+      RC = xmlTextWriterWriteElement(Writer, XmlConstants::MAXVALUE,
+                                     BAD_CAST std::to_string(Max).data());
+      CHECK_RC
+    } else {
+      auto Values = std::get<std::vector<int>>(ValueVariant);
+      // join values to string
+      std::string Str;
+      std::for_each(std::begin(Values), std::end(Values) - 1, [&Str](int X) {
+        Str.append(std::to_string(X));
+        Str.push_back(';');
+      });
+      Str.append(std::to_string(Values.back()));
+
+      RC = xmlTextWriterWriteElement(Writer, XmlConstants::VALUES,
+                                     BAD_CAST Str.data());
+      CHECK_RC
+    }
+  }
+
+  // location?
+  if (FeatureSourceRange *Fsr = Feature1.getFeatureSourceRange()) {
+    RC = writeSourceRange(Writer, *Fsr);
+    CHECK_RC
+  }
+
+  RC = xmlTextWriterEndElement(Writer); // CONFIGURATIONOPTION
+  return RC;
+}
+
+int FeatureModelXmlWriter::writeSourceRange(xmlTextWriterPtr Writer,
+                                            FeatureSourceRange &Location) {
+  int RC;
+  RC = xmlTextWriterStartElement(Writer, XmlConstants::LOCATION);
+  CHECK_RC
```

Comment:
```suggestion
  CHECK_RC

```

Maybe having a new line after every CHECK or a comment before the next parse line makes it a bit more readable